File: 0792-binary-search/solution.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        const int size = nums.size();
        int lo = 0, hi = size - 1;

        while (lo <= hi) {
            int mid = lo + (hi - lo)/2;
            if (nums[mid] == target) {
                return mid;
            } else if (nums[mid] < target) {
                lo = mid + 1;
            } else {
                hi = mid - 1;
            }
        }

        return -1;
    }
};

File: 0792-binary-search/solution.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        lo = 0
        hi = len(nums) - 1
        
        while hi >= lo:
            mid = (lo + hi)//2
            
            if nums[mid] == target:
                return mid
            elif nums[mid] > target:
                hi = mid - 1
            else:
                lo = mid + 1
        return -1
